docs/background/geo.md

```diff
@@ -0,0 +1,235 @@
+    ...
+    
+```
+
+It is not required to implement `invert`, in this case it is better to use `NoInvertProjector` which throws error if somebody will use invert function, when projector doesn't actually support it.
+
+### Composed projectors
+
+It is possible to compose two projectors ino one by `ComposedProjector`. For example it is used in `ProjectorProjection` to combine scale projector, rotate projector and translate projector
+
+### Conditional projectors
+
+Some projections like `ConicConformal` use different projectors for different `lambda` and `phi`. Special `ConditionalProjector` provides API to select nexted projector before `project` and `invert` operations
+
+### Performance
+
+Point project and invert is easy to understand and write, but it is have one big issue: low performance due to a lot of `doubleArray` creation during transformation.
+
+To remove this bottleneck `Projector` have additional methods
+
+```
+    fun projectLambda(lambda: Double, phi: Double): Double
+
+    fun projectPhi(lambda: Double, phi: Double): Double
+    
+    fun invertLambda(lambda: Double, phi: Double): Double
+
+    fun invertPhi(lambda: Double, phi: Double): Double
+
+```
+
+Internal code sometimes use point and sometimes use axys transformations (whatever is best for performance in current state).
+
+Internal Projectors ovverides both all 6 methods (2 for point project/invert and 4 for separated project/invert by axys).
+
+There are `SimpleProjector` which have bad performance but require to override only point project/invert functions
+
+## Projection
+
+`Projection` is container for any `Projector` which provides additional transformations like:
+
+* `scale`
+* `translate`
+* `rotate`
+* `precision`
+* `center`
+
+And clipping
+
+* `preClip`
+* `postClip`
+
+`Projection` provides `stream` method to apply all transformations and receive `Path` with transformed and clipped original `GeoJSON` object
+
+Base implementation for `Projection` is `ProjectorProjection` which actually add additional transformations to given `Projector`
+
+### Transformations API
+
+Projection supports two ways to add additional transformations - by **point** and by **axys**
+For example `Translate` (same for other APIs):
+
+```
+var translateX: Double // x axys
+
+var translateY: Double // y axys
+
+fun translate(x: Double, y: Double) // point
+
+```
+
+Point transformation is fast and usually used in internal calculations.
+Axys transformations is easy to use in client code, for example when you want to change only one axys
+
+
+### Clipping API
+
+`Projection` provides `preClip` and `postClip` 
+
+Original d3-geo API contains `angleClip`, `extentClip`, `antimeridianClip`, `noClip` are implemented as extesions, because it is actually special variants for `preClip` and `postClip`
+
+Geo module provides fit clipping extesions, which helps to clip projeciton to given `GeoJsonObject` area:
+
+```
+fun Projection.fitExtent(extent: Extent, geo: GeoJsonObject): Projection
```

Comment:
We are not in the reference documentation: we don't want the signature but the concepts. What fits functions do?